sonic_platform_base/card_base.py

```diff
@@ -0,0 +1,80 @@
+#
+# card_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a card in SONiC chassis
+#
+
+class CardBase(object):
+    """
+    Abstract base class for implementing a platform specific class to
+    represent a control-card, line-card or fabric-card of a chassis
+    """
+
+    # Possible card types
+    MODULE_TYPE_CONTROL = "CONTROL-CARD"
+    MODULE_TYPE_LINE    = "LINE-CARD"
+    MODULE_TYPE_FABRIC  = "FABRIC-CARD"
+
+    def get_name(self):
+        """
+        Retrieves the name of the card
+
+        Returns:
+            string: The name of the card
+        """
+        raise NotImplementedError
+
+    def get_description(self):
+        """
+        Retrieves the description of the card
+
+        Returns:
+            string: The description of the card
+        """
+        raise NotImplementedError
+
+    def get_instance(self):
+        """
+        Retrieves the slot where the card is present
+
+        Returns:
+            string: slot representation, usually number
+        """
+        raise NotImplementedError
+
+    def get_type(self):
+        """
+        Retrieves the type of the card
+
+        Returns:
+            string: module-type MODULE_TYPE_CONTROL, MODULE_TYPE_LINE etc
+        """
+        raise NotImplementedError
+
+    def get_status(self):
+        """
+        Retrieves the status of the card
+
+        Returns:
+            string: The status-string of the card
+        """
+        raise NotImplementedError
+
+    def reboot_card(self):
+        """
+        Request to reboot/reset the card
+
+        Returns:
+            bool: True if the request has been successful, False if not
```

Comment:
please clarify in the description what does it mean by successful? does it mean the line card reboot into a os?

Reply:
is there difference between reboot and reset? if not, why do we need reboot and reset in the description?

Reply:
There is no difference between reboot/reset here. It will mean the Module will go offline. The vendor specific API implementation can return successful if the underlying shutdown/powerdown command was issued.